strawberryfields/gbs/similarity.py

```diff
@@ -201,3 +203,42 @@ def event_to_sample(photon_number: int, max_count_per_mode: int, modes: int) ->
             available_modes.remove(j)
 
     return sample
+
+
+def feature_vector_sampling(
+    samples: list, event_photon_numbers: list, max_count_per_mode: int = 2
+) -> np.ndarray:
+    """Calculates feature vector with respect to input samples.
+
+    The feature vector is composed of event probabilities, with all events having a maximum
+    photon count in each mode of ``max_count_per_mode``. Events are then described by their total
+    photon number, and those chosen as part of the feature vector can be specified through
+    ``event_photon_numbers``. Probabilities are reconstructed by measuring the occurrence of
+    events in the input ``samples``.
+
+    **Example usage**:
+
+    >>> feature_vector_sampling(samples, [2, 4, 6], 1)
+    (0.34, 0.08, 0.01)
+
+    Args:
+        samples (list[list[int]]): a list of samples
+        event_photon_numbers (list[int]): a list of events described by their total photon number
+        max_count_per_mode (int): maximum number of photons per mode in every event

```

Comment:
Does this read better?
```suggestion
        max_count_per_mode (int): maximum number of photons per mode for all events
```